test/integration/segmentation_camera_plugin.cc

```diff
@@ -0,0 +1,305 @@
+  sdf::init(doc);
+  ASSERT_TRUE(sdf::readFile(path, doc));
+  ASSERT_NE(nullptr, doc->Root());
+  ASSERT_TRUE(doc->Root()->HasElement("model"));
+  auto modelPtr = doc->Root()->GetElement("model");
+  ASSERT_TRUE(modelPtr->HasElement("link"));
+  auto linkPtr = modelPtr->GetElement("link");
+  ASSERT_TRUE(linkPtr->HasElement("sensor"));
+  auto sensorPtr = linkPtr->GetElement("sensor");
+  ASSERT_TRUE(sensorPtr->HasElement("camera"));
+  auto cameraPtr = sensorPtr->GetElement("camera");
+  ASSERT_TRUE(cameraPtr->HasElement("image"));
+  auto imagePtr = cameraPtr->GetElement("image");
+
+  int width = imagePtr->Get<int>("width");
+  int height = imagePtr->Get<int>("height");
+
+  // Setup ign-rendering with an empty scene
+  auto *engine = ignition::rendering::engine(_renderEngine);
+  if (!engine)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' is not supported" << std::endl;
+    return;
+  }
+
+  ignition::rendering::ScenePtr scene = engine->CreateScene("scene");
+  BuildScene(scene);
+
+  ignition::sensors::Manager mgr;
+
+  sdf::Sensor sdfSensor;
+  sdfSensor.Load(sensorPtr);
+
+  std::string type = sdfSensor.TypeStr();
+  EXPECT_EQ(type, "segmentation_camera");
+
+  ignition::sensors::SegmentationCameraSensor *sensor =
+    mgr.CreateSensor<ignition::sensors::SegmentationCameraSensor>(sdfSensor);
+
+  EXPECT_NE(sensor, nullptr);
+  sensor->SetScene(scene);
+
+  EXPECT_EQ(width, sensor->ImageWidth());
+  EXPECT_EQ(height, sensor->ImageHeight());
+
+  auto camera = sensor->SegmentationCamera();
+  EXPECT_NE(camera, nullptr);
+
+  uint32_t backgroundLabel = 23;
+  camera->SetSegmentationType(SegmentationType::Semantic);
+  camera->EnableColoredMap(false);
+  camera->SetBackgroundLabel(backgroundLabel);
+  camera->SetLocalPosition(0.0, 0.0, 0.0);
+  camera->SetLocalRotation(0.0, 0.0, 0.0);
+
+  // Get the Msg
+  std::string topic =
+    "/test/integration/SegmentationCameraPlugin_imagesWithBuiltinSDF";
+  WaitForMessageTestHelper<ignition::msgs::Image> helper(topic);
+
+  // Update once to create image
+  mgr.RunOnce(std::chrono::steady_clock::duration::zero());
+
+  EXPECT_TRUE(helper.WaitForMessage()) << helper;
+
+  // subscribe to the segmentation camera topic
+  ignition::transport::Node node;
+  node.Subscribe(topic, &OnNewSegmentationFrame);
+
+  // wait for a few segmentation camera frames
+  mgr.RunOnce(std::chrono::steady_clock::duration::zero(), true);
+
+  // wait for a new frame
+  WaitForNewFrame();
+
+  // get the center of each box, the percentages locates the center
+  math::Vector2d leftProj(width * 0.25, height * 0.5);
+  math::Vector2d rightProj(width * 0.75, height * 0.5);
+  math::Vector2d middleProj(width * 0.5, height * 0.5);
+
+  // get their index in the buffer
+  uint32_t leftIndex = (leftProj.Y() * width + leftProj.X()) * 3;
+  uint32_t rightIndex = (rightProj.Y() * width + rightProj.X()) * 3;
+  uint32_t middleIndex = (middleProj.Y() * width + middleProj.X()) * 3;
```

Comment:
```suggestion
  // get their index in the buffer
  const uint32_t imgChannels = 3;
  uint32_t leftIndex = (leftProj.Y() * width + leftProj.X()) * imgChannels;
  uint32_t rightIndex = (rightProj.Y() * width + rightProj.X()) * imgChannels;
  uint32_t middleIndex = (middleProj.Y() * width + middleProj.X()) * imgChannels;
```
We should refrain from using "magic numbers" if possible